### PR TITLE
Do not allow to mint dollar when collateral is equal to zero

### PR DESCRIPTION
**Pull request by gitcoindev**
Resolves https://github.com/sherlock-audit/2023-12-ubiquity-judging/issues/207

### CODE REVIEW

File: packages/contracts/src/dollar/libraries/LibUbiquityPool.sol [hunk 1]
```diff
@@ -368,6 +368,7 @@ library LibUbiquityPool {
 
         // get amount of collateral for minting Dollars
         collateralNeeded = getDollarInCollateral(collateralIndex, dollarAmount);
+        require(collateralNeeded > 0, "Cannot mint with zero collateral");
```
Comment (rndquu) [inline packages/contracts/src/dollar/libraries/LibUbiquityPool.sol:371]: @gitcoindev Could you add a unit test?

Comment (gitcoindev) [reply packages/contracts/src/dollar/libraries/LibUbiquityPool.sol:371]: @rndquu yes, will add one.

File: packages/contracts/test/diamond/facets/UbiquityPoolFacet.t.sol [hunk 2]
```diff
@@ -329,6 +329,27 @@ contract UbiquityPoolFacetTest is DiamondTestSetup {
         );
     }
 
+    function testMintDollar_ShouldRevert_IfZeroCollateralAvailable() public {
```
Comment (molecula451) [inline packages/contracts/test/diamond/facets/UbiquityPoolFacet.t.sol:332]: 👀 ,

File: packages/contracts/src/dollar/libraries/LibUbiquityPool.sol (hunk 1 above)
Comment (gitcoindev) [reply packages/contracts/src/dollar/libraries/LibUbiquityPool.sol:371]: done, unit test implemented and passing

File: packages/contracts/test/diamond/facets/UbiquityPoolFacet.t.sol (hunk 2 above)
Comment (gitcoindev) [reply packages/contracts/test/diamond/facets/UbiquityPoolFacet.t.sol:332]: @molecula451 this is related to https://github.com/ubiquity/ubiquity-dollar/pull/892#discussion_r1476177155

**Comment by molecula451** [comment]
is this ready? @gitcoindev

**Comment by gitcoindev** [comment]
> is this ready? @gitcoindev

Yes, the unit test expects revert, which happens, and passes.

**Review by molecula451 — APPROVED** [review]

**Review by rndquu — APPROVED** [review]

**Pull request merged**